gdscript-eglot.el

```diff
@@ -42,6 +42,25 @@
   "The version of godot in use."
   :type 'string)
 
+(defun gdscript-eglot--get-config-dir ()
+  (pcase system-type
+    ('darwin "~/Library/Application Support/Godot/")
+    ('windows-nt (substitute-in-file-name "$APPDATA/Godot/"))
+    ('gnu/linux (file-name-concat
+                 (or (getenv "XDG_CONFIG_HOME") "~/.config/")
+                 "godot"))))
+
+(defun gdscript-eglot--extract-port (editor-settings-file)
+  (when (file-exists-p editor-settings-file)
+    (with-temp-buffer
+      (insert-file-contents editor-settings-file)
+      (when (re-search-forward
+             (rx "network/language_server/remote_port"
+                 (* space) ?= (* space)
+                 (group (+ digit)))
+             nil t)
+        (string-to-number (match-string 1))))))
```

Comment:
Can you add documentation to these functions? Thanks!

Reply:
I've added documentation for `gdscript-eglot--get-config-dir` and `gdscript-eglot--extract-port` functions.